### PR TITLE
Double click headers mouse move

### PR DESCRIPTION
**Pull request by davidfig**
- [x] fixes bug where you can click the bottom-right corner of a the header, move the mouse, and still complete the double click to resize the column
- [x] fixed a nasty bug where automatically resizing the column would result in bad rendering of columns. also fixed the same bug with undoing that change 
- [x] changed the minimum column size from 20px (which is the minimum a user can shrink the heading) to 200px (which is the default column width)
- [x] read-only can no longer resize or auto resize headers

### CODE REVIEW

File: quadratic-core/src/controller/sheet_offsets.rs [hunk 1]
```diff
@@ -39,6 +39,29 @@ impl GridController {
             TransactionSummary::default()
         }
     }
+
+    pub fn commit_single_resize(
```
Comment (ddimaria) [inline quadratic-core/src/controller/sheet_offsets.rs:43]: Can you add a TODO here to test this function?

**Review by ddimaria — APPROVED** [review]
Code looks good.

File: quadratic-core/src/controller/sheet_offsets.rs (hunk 1 above)
Comment (davidkircos) [reply quadratic-core/src/controller/sheet_offsets.rs:43]: Is now a good time to just add some tests?

Comment (davidfig) [reply quadratic-core/src/controller/sheet_offsets.rs:43]: added some tests

**Pull request merged**